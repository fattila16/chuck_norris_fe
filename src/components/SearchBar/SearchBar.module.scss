@import 'src/assets/scss/color-palette';
@import 'src/assets/scss/vars';

.input {
    border: none;
    background-color: map-get($active-theme , white );
    font-size: 18px;
    font-weight: 300;
    width: 25vw;
    padding: 10px;
    border-radius: 20px;
}