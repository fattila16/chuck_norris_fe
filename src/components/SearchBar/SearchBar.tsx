import React, { ChangeEvent } from 'react';
import css from './SearchBar.module.scss';

interface InputProps {
    onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
    value?: string;
    placeHolder?: string;
}
const SearchBar: React.FC<InputProps> = (props: InputProps) => {
    const { onChange, value, placeHolder } = props;
    return (
        <>
            <input className={css.input} type="text" onChange={onChange} value={value} placeholder={placeHolder} />
        </>
    );
};

export default SearchBar;
