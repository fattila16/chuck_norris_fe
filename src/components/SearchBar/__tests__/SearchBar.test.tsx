import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../SearchBar';

describe('[SearchBar]', () => {
    it('Should render SearchBar component without error', () => {
        const component = shallow(<SearchBar />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
