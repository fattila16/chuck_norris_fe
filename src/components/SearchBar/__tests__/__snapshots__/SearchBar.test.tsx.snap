// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SearchBar] Should render SearchBar component without error 1`] = `
Array [
  <React.Fragment>
    <input
      className="input"
      type="text"
    />
  </React.Fragment>,
]
`;
